xtable-utilities/src/main/java/org/apache/xtable/utilities/RunSync.java

```diff
@@ -115,66 +116,116 @@ public class RunSync {
+      Configuration hadoopConf,
+      ConversionSourceProvider conversionSourceProvider) {
+    ConversionController conversionController = new ConversionController(hadoopConf);
+    for (DatasetConfig.Table table : datasetConfig.getDatasets()) {
+      log.info(
+          "Running sync for basePath {} for following table formats {}",
+          table.getTableBasePath(),
+          tableFormatList);
+      Properties sourceProperties = new Properties();
+      if (table.getPartitionSpec() != null) {
+        sourceProperties.put(
+            HudiSourceConfig.PARTITION_FIELD_SPEC_CONFIG, table.getPartitionSpec());
+      }
 
-    if (cmd.hasOption(CONTINUOUS_MODE)) {
-      ScheduledExecutorService executorService = Executors.newScheduledThreadPool(1);
-      long intervalInSeconds = Long.parseLong(cmd.getOptionValue(CONTINUOUS_MODE_INTERVAL, "5"));
-      executorService.scheduleAtFixedRate(
-          () -> {
-            try {
-              runSync(cmd);
-            } catch (IOException ex) {
-              log.error("Sync operation failed", ex);
-            }
-          },
-          0,
-          intervalInSeconds,
-          TimeUnit.SECONDS);
-      while (!Thread.currentThread().isInterrupted()) {
-        try {
-          Thread.sleep(1000);
-        } catch (InterruptedException ex) {
-          log.debug("Received interrupt signal");
-          Thread.currentThread().interrupt();
-          break;
-        }
+      SourceTable sourceTable =
+          sourceTableBuilder(table, icebergCatalogConfig, datasetConfig, sourceProperties);
+      List<TargetTable> targetTables =
+          targetTableBuilder(table, icebergCatalogConfig, tableFormatList);
+      ConversionConfig conversionConfig =
+          ConversionConfig.builder()
+              .sourceTable(sourceTable)
+              .targetTables(targetTables)
+              .syncMode(SyncMode.INCREMENTAL)
+              .build();
+      try {
+        conversionController.sync(conversionConfig, conversionSourceProvider);
+      } catch (Exception e) {
+        log.error(String.format("Error running sync for %s", table.getTableBasePath()), e);
       }
-      executorService.shutdownNow();
-    } else {
-      runSync(cmd);
     }
   }
-
-  private static void runSync(CommandLine cmd) throws IOException {
-    DatasetConfig datasetConfig;
-    try (InputStream inputStream =
-        Files.newInputStream(Paths.get(cmd.getOptionValue(DATASET_CONFIG_OPTION)))) {
-      datasetConfig = YAML_MAPPER.readValue(inputStream, DatasetConfig.class);
+  
+  static DatasetConfig getDatasetConfig(String datasetConfigPath) throws IOException {
+    // Initialize DatasetConfig
+    DatasetConfig datasetConfig = new DatasetConfig();
+
+    try (InputStream inputStream = Files.newInputStream(Paths.get(datasetConfigPath))) {
+      ObjectReader objectReader = YAML_MAPPER.readerForUpdating(datasetConfig);
+      objectReader.readValue(inputStream);
     }
+    return datasetConfig;
```

Comment:
```suggestion
    try (InputStream inputStream =
        Files.newInputStream(Paths.get(cmd.getOptionValue(DATASET_CONFIG_OPTION)))) {
      return YAML_MAPPER.readValue(inputStream, DatasetConfig.class);
    }
```

Reply:
@the-other-tim-brown this change was requested by @rangareddy we are not passing cmd everytime to these methods. Instead we are creating config paths just once using cmd and using them later on. So I have accepted only this line for change `return YAML_MAPPER.readValue(inputStream, DatasetConfig.class);`